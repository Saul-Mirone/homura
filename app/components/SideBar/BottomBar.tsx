import React from 'react';
import { PlusIcon, RefreshIcon } from '../Icon';
import { IconContainer } from '../LogoIcon';

export enum Step {
  EnterUrl,
  EnterName,
}

export type BottomBarProps = {
  step: Step | null;
  onClickPlus: () => void;
  onClickSync: () => void;
  render: (step: Step) => JSX.Element;
  refreshing: boolean;
};

export const BottomBar: React.FC<BottomBarProps> = ({
  step,
  onClickPlus,
  onClickSync,
  render,
  refreshing,
}) => (
  <div className="relative flex justify-between text-gray-300">
    {step !== null && (
      <div className="absolute bottom-0 inset-x-0 z-10">{render(step)}</div>
    )}
    <IconContainer onClick={onClickPlus}>
      <PlusIcon />
    </IconContainer>
    <IconContainer onClick={onClickSync}>
      <RefreshIcon spin={refreshing} />
    </IconContainer>
  </div>
);
