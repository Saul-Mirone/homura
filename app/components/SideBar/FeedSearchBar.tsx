import React from 'react';
import { SearchCircleFilledIcon } from '../Icon';
import { IconContainer } from '../LogoIcon';

export type FeedSearchBarProps = {
  link: string;
  loading: boolean;
  onLinkChange: (link: string) => void;
  onSearch: (link: string) => void;
  onCancel: () => void;
};

export const FeedSearchBar: React.FC<FeedSearchBarProps> = ({
  onSearch,
  link,
  onLinkChange,
  onCancel,
  loading,
}) => {
  const inputEl = React.useRef<HTMLInputElement>(null);
  React.useEffect(() => {
    inputEl.current?.focus();
  }, []);

  return (
    <div className="flex bg-gray-300 text-gray-800">
      <div className="flex-1">
        <input
          ref={inputEl}
          value={link}
          onChange={(e) => onLinkChange(e.target.value)}
          onBlur={() => {
            onCancel();
          }}
          className="w-full p-2 text-xs bg-transparent"
          placeholder="Feed URL"
        />
      </div>
      <IconContainer
        className={`transition duration-300 ${
          !loading
            ? 'hover:bg-gray-700 hover:text-gray-300 text-gray-700'
            : 'bg-gray-600 text-gray-300'
        }`}
        disabled={loading}
        onClick={() => {
          if (loading) return;
          onSearch(link);
        }}
      >
        <SearchCircleFilledIcon />
      </IconContainer>
    </div>
  );
};
