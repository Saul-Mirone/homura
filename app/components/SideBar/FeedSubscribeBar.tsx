import React from 'react';
import { CheckCircleFilledIcon } from '../Icon';
import { IconContainer } from '../LogoIcon';

export type FeedSubscribeBarProps = {
  link: string;
  name: string;
  onNameChange: (name: string) => void;
  onConfirm: (name: string) => void;
  onCancel: () => void;
};

export const FeedSubscribeBar: React.FC<FeedSubscribeBarProps> = ({
  link,
  name,
  onNameChange,
  onConfirm,
  onCancel,
}) => {
  const inputEl = React.useRef<HTMLInputElement>(null);
  React.useEffect(() => {
    inputEl.current?.focus();
  }, []);

  return (
    <>
      <div className="p-2 text-xs truncate">{link}</div>
      <div className="flex bg-gray-300 text-gray-800">
        <div className="flex-1">
          <input
            ref={inputEl}
            value={name}
            onChange={(e) => onNameChange(e.target.value)}
            onBlur={() => {
              onCancel();
            }}
            className="w-full p-2 text-xs bg-transparent"
            placeholder="Save as Name"
          />
        </div>
        <IconContainer
          onClick={() => onConfirm(name)}
          className="transition duration-300 text-gray-700 hover:bg-gray-700 hover:text-gray-300"
        >
          <CheckCircleFilledIcon />
        </IconContainer>
      </div>
    </>
  );
};
