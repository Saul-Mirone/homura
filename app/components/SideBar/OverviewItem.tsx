import React from 'react';
import { Preset } from '../../constants/Preset';
import { SideBarItem } from './SideBarItem';

export type OverViewItemProps = {
  activeId?: Preset | number;
  target: Preset;
  count: number;
  icon: JSX.Element;
  onClick: (target: Preset) => void;
};

export const OverviewItem: React.FC<OverViewItemProps> = ({
  activeId,
  target,
  icon,
  count,
  onClick,
}) => (
  <SideBarItem
    active={activeId === target}
    icon={icon}
    name={target}
    count={count}
    onClick={() => onClick(target)}
  />
);
