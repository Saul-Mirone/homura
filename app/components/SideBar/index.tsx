import React from 'react';

export type SideBarProps = {
  overview: JSX.Element;
  bottom: JSX.Element;
  onClick: () => void;
};

export const SideBar: React.FC<SideBarProps> = ({
  overview,
  bottom,
  children,
  onClick,
}) => {
  return (
    <div
      role="presentation"
      onClick={onClick}
      className="w-1/6 bg-gray-700 pt-10 flex flex-col h-screen select-none"
    >
      <div className="flex-1 thin-scroll">
        {overview}
        <div className="pt-10" />
        {children}
      </div>
      {bottom}
    </div>
  );
};
