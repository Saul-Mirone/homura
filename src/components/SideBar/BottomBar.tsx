import React from 'react';
import { PlusIcon, RefreshIcon } from '../Icon';
import { IconContainer } from '../LogoIcon';

export enum Step {
  EnterUrl,
  EnterName,
}

export type BottomBarProps = {
  step: Step | undefined;
  onClickPlus: () => void;
  onClickSync: () => void;
  render: (step: Step) => JSX.Element;
  loading: boolean;
};

export const BottomBar: React.FC<BottomBarProps> = ({
  step,
  onClickPlus,
  onClickSync,
  render,
  loading,
}) => (
  <div role="toolbar" className="sidebar-bottom-bar">
    {step !== undefined && (
      <div className="sidebar-bottom-bar__step">{render(step)}</div>
    )}
    <IconContainer onClick={onClickPlus}>
      <PlusIcon />
    </IconContainer>
    <IconContainer onClick={onClickSync}>
      <RefreshIcon spin={loading} />
    </IconContainer>
  </div>
);
