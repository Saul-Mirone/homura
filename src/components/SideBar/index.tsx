import React from 'react';
import './style.pcss';

export type SideBarProps = {
    overview: JSX.Element;
    bottom: JSX.Element;
    onClick: () => void;
};

export const SideBar: React.FC<SideBarProps> = ({ overview, bottom, children, onClick }) => (
    <div
        role="menu"
        tabIndex={0}
        aria-label="Sources"
        onClick={onClick}
        onKeyDown={onClick}
        className="sidebar-container"
    >
        <div className="relative flex-1 thin-scroll">
            {overview}
            <div className="pt-10" />
            {children}
        </div>
        {bottom}
    </div>
);
