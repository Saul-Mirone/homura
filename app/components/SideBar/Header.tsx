import React from 'react';
import { Mode } from '../../constants/Mode';
import { Preset } from '../../constants/Preset';
import {
  ArchiveIcon,
  MailFilledIcon,
  StarFilledIcon,
  ViewListIcon,
} from '../Icon';
import { OverviewItem } from './OverviewItem';

export type HeaderProps = {
  mode: Mode;
  count: number;
  onClick: (target: Preset) => void;
  active: Preset | number | null;
};

export const Header: React.FC<HeaderProps> = ({
  active,
  mode,
  count,
  onClick,
}) => {
  switch (mode) {
    case Mode.Starred:
      return (
        <OverviewItem
          testId="source-list-preset-starred"
          target={Preset.Starred}
          icon={<StarFilledIcon />}
          activeId={active}
          count={count}
          onClick={onClick}
        />
      );
    case Mode.Unread:
      return (
        <OverviewItem
          testId="source-list-preset-unread"
          target={Preset.Unread}
          icon={<MailFilledIcon />}
          activeId={active}
          count={count}
          onClick={onClick}
        />
      );
    default:
    case Mode.All:
      return (
        <>
          <OverviewItem
            testId="source-list-preset-all"
            target={Preset.All}
            icon={<ViewListIcon />}
            activeId={active}
            count={count}
            onClick={onClick}
          />
          <OverviewItem
            testId="source-list-preset-archive"
            target={Preset.Archive}
            icon={<ArchiveIcon />}
            activeId={active}
            count={0}
            onClick={onClick}
          />
        </>
      );
  }
};
