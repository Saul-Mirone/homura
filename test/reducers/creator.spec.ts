import { AnyAction } from 'redux';
import { Step } from '../../app/components/SideBar/BottomBar';
import {
  creatorReducer,
  reset,
  setLink,
  setName,
  stepToEnterName,
  stepToEnterUrl,
} from '../../app/features/creator/creatorSlice';

describe('reducers', () => {
  describe('creator', () => {
    it('should handle initial state', () => {
      expect(creatorReducer(undefined, {} as AnyAction)).toMatchSnapshot();
    });

    it('should handle stepToEnterUrl', () => {
      expect(
        creatorReducer(
          { step: null, link: '', name: '' },
          { type: stepToEnterUrl }
        )
      ).toMatchSnapshot();
    });

    it('should handle stepToEnterName', () => {
      expect(
        creatorReducer(
          { step: null, link: '', name: '' },
          { type: stepToEnterName }
        )
      ).toMatchSnapshot();
    });

    it('should handle reset', () => {
      expect(
        creatorReducer(
          { step: Step.EnterUrl, link: 'link', name: 'name' },
          { type: reset }
        )
      ).toMatchSnapshot();
    });

    it('should handle setLink', () => {
      expect(
        creatorReducer(
          { step: null, link: '', name: '' },
          { type: setLink, payload: 'link' }
        )
      ).toMatchSnapshot();
    });

    it('should handle setName', () => {
      expect(
        creatorReducer(
          { step: null, link: '', name: '' },
          { type: setName, payload: 'name' }
        )
      ).toMatchSnapshot();
    });

    it('should handle unknown action type', () => {
      expect(
        creatorReducer({ step: null, link: '', name: '' }, { type: 'unknown' })
      ).toMatchSnapshot();
    });
  });
});
