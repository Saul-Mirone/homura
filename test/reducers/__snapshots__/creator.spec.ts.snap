// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers creator should handle initial state 1`] = `
Object {
  "link": "",
  "name": "",
  "step": null,
}
`;

exports[`reducers creator should handle reset 1`] = `
Object {
  "link": "",
  "name": "",
  "step": null,
}
`;

exports[`reducers creator should handle setLink 1`] = `
Object {
  "link": "link",
  "name": "",
  "step": null,
}
`;

exports[`reducers creator should handle setName 1`] = `
Object {
  "link": "",
  "name": "name",
  "step": null,
}
`;

exports[`reducers creator should handle stepToEnterName 1`] = `
Object {
  "link": "",
  "name": "",
  "step": 1,
}
`;

exports[`reducers creator should handle stepToEnterUrl 1`] = `
Object {
  "link": "",
  "name": "",
  "step": 0,
}
`;

exports[`reducers creator should handle unknown action type 1`] = `
Object {
  "link": "",
  "name": "",
  "step": null,
}
`;
